sonar-dotnet-shared-library/src/test/java/org/sonar/plugins/dotnet/tests/ScannerFileServiceTest.java

```diff
@@ -199,8 +200,9 @@ public void getAbsolutePath_with_no_deterministic_path_in_windows_path_returns_e
     verify(fs, never()).predicates();
     assertThat(logTester.logs(Level.TRACE)).isEmpty();
     assertThat(logTester.logs(Level.DEBUG)).hasSize(1);
-    assertThat(logTester.logs(Level.DEBUG).get(0)).isEqualTo("Did not find deterministic source path in 'C:\\_\\some\\path\\file.cs'." +
-      " Will skip this coverage entry. Verify sonar.sources in .sonarqube\\out\\sonar-project.properties.");
+    assertThat(logTester.logs(Level.DEBUG).get(0)).isEqualTo("The file 'C:\\_\\some\\path\\file.cs' is not indexed or "
+      + "does not have the supported language and does not have deterministic build path. "
```

Comment:
The combination of the conditions in the message isn't clear to me:
- is it `not indexed OR (not supported language AND not deterministic path)`
- or is it `(not indexed OR not supported language) AND not deterministic path`?

Reply:
the latter. Not sure how to make it more clear :D

Reply:
I'm not sure either.
How about putting the reasons in separate lines?
I'll leave it up to you.

```
The coverage entry for 'C:\_\some\path\file.cs' will be skipped for the following reasons:
- the file is not indexed or does not have the supported language
- the file does not have a deterministic build path 
```